presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -194,4 +218,22 @@ public Symbol getRight()
             return right;
         }
     }
+
+    public boolean isDistributed()
+    {
+        return method == DISTRIBUTED;
+    }
+
+    public static boolean canDistributeJoin(Session session, PlanNode rightNode, List<JoinNode.EquiJoinClause> equiJoinClauses, JoinNode.Type type)
```

Comment:
Good point.  I'll put all of this in it's own optimizer.

I was also unclear about the distributed_joins session property-- Does that mean never do broadcast joins when that's set, or does it mean never do distributed joins when it's not set.  Practically there is no difference now, but it becomes relevant with the upcoming optimizer to automatically choose between the two.